tests/python/pants_test/base/test_hash_utils.py

```diff
@@ -82,3 +86,83 @@ def check_bad_spec(spec):
     check_bad_spec('/')
     check_bad_spec('foo/1')
     check_bad_spec('1/foo')
+
+
+class CoercingJsonEncodingTest(unittest.TestCase):
+  def _coercing_json_encode(self, o, digest=None):
+    return json.dumps(o, cls=CoercingEncoder)
+
+  def test_normal_object_encoding(self):
+    self.assertEqual(self._coercing_json_encode({}), '{}')
+    self.assertEqual(self._coercing_json_encode(()), '[]')
+    self.assertEqual(self._coercing_json_encode([]), '[]')
+    self.assertEqual(self._coercing_json_encode(set([])), '[]')
```

Comment:
`set([]) == set()` is `True`

```suggestion
    self.assertEqual(stable_json_sha1(set()), '97d170e1550eee4afc0af065b78cda302a97674c')
```